solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetContentTypeFilter.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2017 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.data.ingest.solr.filter;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.unc.lib.dl.data.ingest.solr.exception.IndexingException;
+import edu.unc.lib.dl.data.ingest.solr.indexing.DocumentIndexingPackage;
+import edu.unc.lib.dl.fcrepo4.BinaryObject;
+import edu.unc.lib.dl.fcrepo4.ContentObject;
+import edu.unc.lib.dl.fcrepo4.FileObject;
+import edu.unc.lib.dl.fcrepo4.WorkObject;
+import edu.unc.lib.dl.search.solr.util.ContentCategory;
+
+/**
+ * Assigns content-type field for a Solr record. The field contains a category (e.g., "image")
+ * and an extension (e.g., ".jpg") for binary content that is contained by the represented object.
+ * 
+ * @author harring
+ *
+ */
+public class SetContentTypeFilter implements IndexDocumentFilter {
+	
+	private static final Logger log = LoggerFactory.getLogger(SetContentTypeFilter.class);
+	private static final Pattern EXTENSION_REGEX = Pattern.compile("^[^\\n]*[^.]\\.(\\d*[a-zA-Z][a-zA-Z0-9]*)[\"'{}, _\\-`()*]*$");
+	private static final int EXTENSION_LIMIT = 8;
+	
+	private Properties mimetypeToExtensionMap;
+	private Properties contentTypeProperties;
+	
+	public SetContentTypeFilter() {
```

Comment:
Looking at this (and i realize this is my own crappy code), its really a configuration error for the files to fail to load, and we really don't want to be eating that and just putting something in the log.  Could you either remove the try block and make this constructor throw the IOException, or rethrow an exception with a little extra context?  Up to you which you prefer.